--- original/default
+++ build/default
@@ -183,7 +183,7 @@
 {#- ===== SETUP ===== -#}
 {%- set ns = namespace(prev_message_type=None, prev_non_tool_role=None) -%}
 {%- set loop_messages = messages -%}
-{%- set enable_thinking = enable_thinking | default(false) -%}
+{%- set enable_thinking = false -%}
 {%- set preserve_thinking = preserve_thinking | default(false) -%}
 {{- bos_token -}}
 {#- Handle System/Tool Definitions Block -#}
